File: src/erl/exceptions.erl
-module(exceptions).
-export([start/0]).

doStuff(InError) ->
    if
        not InError -> io:fwrite("Doing stuff!~n");
        true -> throw("An error occured!")
    end.

start() -> 
    try
        doStuff(false),
        doStuff(true),
        doStuff(false) % not reached
    catch
        throw:Message -> io:fwrite("ERROR: " ++ Message)
    end,

    erlang:halt().

File: src/ml/exceptions.ml
let doStuff inError =
    if inError
    then raise (Failure "An error occured!")
    else print_string "Doing stuff!\n";;

try
    doStuff false;
    doStuff true;
    doStuff false;
with
    Failure message -> Printf.printf "ERROR: %s\n" message;;
    